src/getCartReceptors.C

```diff
@@ -167,30 +235,35 @@ void MeshBlock::getCartReceptors(CartGrid *cg,parallelComm *pc)
   // if these were already allocated
   // get rid of them
   //
-  if (xsearch) TIOGA_FREE(xsearch);
-  if (isearch) TIOGA_FREE(isearch);
   if (donorId) TIOGA_FREE(donorId);
+  if (tagsearch) TIOGA_FREE(tagsearch);
+  if (isearch) TIOGA_FREE(isearch);
   if (rst) TIOGA_FREE(rst);
+  if (res_search) TIOGA_FREE(res_search);
+  if (xsearch) TIOGA_FREE(xsearch);
   //
-  xsearch=(double *)malloc(sizeof(double)*3*nsearch);
-  isearch=(int *)malloc(3*sizeof(int)*nsearch);
   donorId=(int *)malloc(sizeof(int)*nsearch);
+  tagsearch=(int *)malloc(sizeof(int)*nsearch);
+  isearch=(int *)malloc(3*sizeof(int)*nsearch);
+  res_search=(double *)malloc(sizeof(double)*nsearch);
+  xsearch=(double *)malloc(sizeof(double)*3*nsearch);
   rst=(double *) malloc(sizeof(double)*3*nsearch);
   //
   dataPtr=head->next;
-  m=n=0;
+  k=l=m=n=0;
   while(dataPtr!=NULL)
-    {
-      for(j=2;j<dataPtr->intDataSize;j++)
-	{
-	  isearch[m++]=dataPtr->intData[0];
-	  isearch[m++]=dataPtr->intData[1];	  
-	  isearch[m++]=dataPtr->intData[j];
-	}
-      for(j=0;j<3*dataPtr->realDataSize;j++)
-	xsearch[n++]=dataPtr->realData[j];
-       dataPtr=dataPtr->next;
-    }
+  {
+    for(j=0;j<dataPtr->intDataSize-1;j++)
```

Comment:
Can you explain the indexing here? and why it changed from `j=2` to `j=0`? I don't understand it.

Reply:
The loop here had a length of 3 and the 1st and 2nd index was being assigned expliciitly instead of via the loop. So I cleaned that up.